SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -12,6 +10,30 @@ DEFINE_LOG_CATEGORY(LogSpatialEventTracer);
 
 namespace SpatialGDK
 {
+TraceQueryPtr ParseOrDefault(const FString& Str, const char* FilterForLog)
+{
+	TraceQueryPtr Ptr;
+	if (Str.Len())
+	{
+		Ptr.Reset(Trace_ParseSimpleQuery(TCHAR_TO_ANSI(*Str)));
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("Applied %s query: %s"), FilterForLog, *Str);
+	}
+	else
+	{
+		Ptr.Reset(Trace_ParseSimpleQuery("true"));
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("No %s specified, using \"true\""), FilterForLog);
+	}
+
+	if (!Ptr.Get())
+	{
+		UE_LOG(LogSpatialEventTracer, Warning, TEXT("The specified %s is invalid, sampling will be disabled. %s"), FilterForLog,
+			   Trace_GetLastError());
+		Ptr.Reset(Trace_ParseSimpleQuery("false"));
```

Comment:
Does this mean nothing will get sampled or everything will get sampled if an invalid query string was entered?

Reply:
One slight oddity in the flow here which is not that important...

if Str length is 0 we reset Ptr using "true". We then go on to check the Ptr and if it is not valid we reset Ptr with ""false". Do we then need to go on and check resetting using "false" was successful :P

Perhaps we only need to check when Str.len > 0

Reply:
Yeah would mean nothing is sampled - I think I'll change some of the defaults around here. You can get yourself into a bit of trouble sampling too much (I took down the NFR analysis step) so it might be better to disable everything unless your query specifies otherwise. 

It also encourages the user to get acquainted with the queries

Reply:
> Do we then need to go on and check resetting using "false" was successful :P

I think I was more worried about the query language changing :D we've got some hardcoded strings here but worker might supply some helper functions so we could rework this like you say

Reply:
Hehe. Yeah some values from worker would be super!